pkg/config/config.go

```diff
@@ -0,0 +1,66 @@
+/*
+ Copyright 2023, NVIDIA CORPORATION & AFFILIATES
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"os"
+
+	"k8s.io/apimachinery/pkg/util/json"
+)
+
+// FromFile reads configuration from the file
+func FromFile(path string) (Config, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return Config{}, fmt.Errorf("failed to read data from the provided file: %v", err)
```

Comment:
IMO, we shouldn't return an empty config. it's enouph to return `nil, fmt.Errorf("failed to read data from the provided file: %v", err)`